From be730431925b61c750794b4a39d8a38367c1bf4a Mon Sep 17 00:00:00 2001
From: Pavankumar Nandeshwar <quic_pnandesh@quicinc.com>
Date: Fri, 21 Jun 2024 23:30:05 +0530
Subject: [PATCH] QSDK: hostapd: Fix interface addition to bridge sequence

Make sure that the network interface is added later to
the bridge. This is needed because by the time networking
module detects a bridge addition, the interface needs
to have all the necessary information ready

Patch-depdendency: None
Signed-off-by: Pavankumar Nandeshwar <quic_pnandesh@quicinc.com>
---
 src/drivers/driver_nl80211.c | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -9202,24 +9202,25 @@ static int wpa_driver_nl80211_if_add(voi
 			return -1;
 		}
 
-		if (bridge &&
-		    i802_check_bridge(drv, new_bss, bridge, ifname) < 0) {
-			wpa_printf(MSG_ERROR, "nl80211: Failed to add the new "
-				   "interface %s to a bridge %s",
-				   ifname, bridge);
+		if (linux_set_iface_flags(drv->global->ioctl_sock, ifname, 1))
+		{
 			if (added)
 				nl80211_remove_iface(drv, ifidx);
 			os_free(new_bss);
 			return -1;
 		}
 
-		if (linux_set_iface_flags(drv->global->ioctl_sock, ifname, 1))
-		{
+		if (bridge &&
+		    i802_check_bridge(drv, new_bss, bridge, ifname) < 0) {
+			wpa_printf(MSG_ERROR, "nl80211: Failed to add the new "
+				   "interface %s to a bridge %s",
+				   ifname, bridge);
 			if (added)
 				nl80211_remove_iface(drv, ifidx);
 			os_free(new_bss);
 			return -1;
 		}
+
 		os_strlcpy(new_bss->ifname, ifname, IFNAMSIZ);
 		os_memcpy(new_bss->addr, if_addr, ETH_ALEN);
 		new_bss->ifindex = ifidx;
